packages/react-component-library/src/components/Timeline/TimelineWeekColumns.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import { differenceInDays, endOfWeek, max, min } from 'date-fns'
+
+import { formatPx, isOdd } from './helpers'
+import { StyledWeekColumn } from './partials/StyledWeekColumn'
+import { StyledWeekColumnsWrapper } from './partials/StyledWeekColumnsWrapper'
+import { StyledWeekColumns } from './partials/StyledWeekColumns'
+import { TimelineContext } from './context'
+import { WEEK_START } from './constants'
+import { withKey } from '../../helpers'
+
+function renderDefaultColumns(
+  index: number,
+  isOddNumber: boolean,
+  offsetPx: string,
+  widthPx: string
+) {
+  return (
+    <StyledWeekColumn
+      isOddNumber={isOddNumber}
+      marginLeft={offsetPx}
+      width={widthPx}
+    />
+  )
+}
+
+interface TimelineWeekColumnProps {
+  renderColumns?: (
+    index: number,
+    isOddNumber: boolean,
+    offsetPx: string,
+    widthPx: string
+  ) => React.ReactElement
+}
+
+export const TimelineWeekColumns: React.FC<TimelineWeekColumnProps> = ({
+  renderColumns,
+}) => (
+  <TimelineContext.Consumer>
+    {({ state: { days, options, weeks } }) => (
+      <StyledWeekColumnsWrapper>
+        <StyledWeekColumns role="presentation" data-testid="timeline-columns">
+          {weeks.map(({ startDate }, index) => {
+            const lastDateDisplayed = min([
```

Comment:
This stuff happens in a few places right? I think this is what you were referring too recently?